internal/store/hpa.go

```diff
@@ -76,6 +78,91 @@ var (
 				}
 			}),
 		},
+		{
+			Name: "kube_hpa_spec_target_metric",
+			Type: metric.Gauge,
+			Help: "The metric specifications used by this autoscaler when calculating the desired replica count.",
+			GenerateFunc: wrapHPAFunc(func(a *autoscaling.HorizontalPodAutoscaler) *metric.Family {
+				targets := make([]*metric.Metric, 0, len(a.Spec.Metrics))
+				for _, m := range a.Spec.Metrics {
+					switch m.Type {
+					case autoscaling.ObjectMetricSourceType:
+						targetValue, ok := m.Object.TargetValue.AsInt64()
+						if ok {
+							targets = append(targets, &metric.Metric{
+								LabelKeys:   targetMetricLabels,
+								LabelValues: []string{m.Object.MetricName, "value"},
+								Value:       float64(targetValue),
+							})
+						}
+						if m.Object.AverageValue != nil {
+							averageValue, ok := m.Object.AverageValue.AsInt64()
+							if ok {
+								targets = append(targets, &metric.Metric{
+									LabelKeys:   targetMetricLabels,
+									LabelValues: []string{m.Object.MetricName, "average"},
+									Value:       float64(averageValue),
+								})
+							}
+						}
+					case autoscaling.PodsMetricSourceType:
+						value, ok := m.Pods.TargetAverageValue.AsInt64()
+						if !ok {
+							continue
+						}
+						targets = append(targets, &metric.Metric{
+							LabelKeys:   targetMetricLabels,
+							LabelValues: []string{m.Pods.MetricName, "average"},
+							Value:       float64(value),
+						})
+					case autoscaling.ResourceMetricSourceType:
+						targets = append(targets, &metric.Metric{
+							LabelKeys:   targetMetricLabels,
+							LabelValues: []string{string(m.Resource.Name), "utilization"},
+							Value:       float64(*m.Resource.TargetAverageUtilization),
+						})
+
+						if m.Resource.TargetAverageValue != nil {
+							value, ok := m.Resource.TargetAverageValue.AsInt64()
+							if !ok {
+								continue
+							}
+							targets = append(targets, &metric.Metric{
+								LabelKeys:   targetMetricLabels,
+								LabelValues: []string{string(m.Resource.Name), "value"},
```

Comment:
Don't forget this change when you address the previous comment.
```suggestion
								LabelValues: []string{string(m.Resource.Name), "average"},
```